docs/handshake.md

```diff
@@ -0,0 +1,1525 @@
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |   Ext Type=SRT_CMD_HSREQ(1)   |         Ext Size {3}          |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                    SRT Version {>=10300h}                     |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                           SRT Flags                           |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |     RcvTsbPdDelay {20..8000}  |   SndTsbPdDelay {20..8000}    |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |    Ext Type=SRT_CMD_KMREQ(3)  |      Ext Size (bytes/4)       |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |0| V{1} PT{2}|          Sign {2029h}           |  Resv {0}  |KK|
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                           KEKI {0}                            |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |  Cipher {2} |    Auth {0}     |     SE {2}    |   Resv1 {0}   |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |           Recv2 {0}           | Slen(bytes)/4 | klen(bytes)/4 |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                           Salt[Slen]                          |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                   Wrap[((KK+1/2)*Klen) + 8]                   |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
+The **Extension Flags** subfield in the `Type` field in a conclusion handshake
+message contains one of these flags:
+
+- `HS_EXT_HSREQ`: defines SRT characteristic data; always present
+- `HS_EXT_KMREQ`: if using encryption, defines encryption block
+- `HS_EXT_CONFIG`: informs about having extra configuration data attached
+
+The above schema shows the HSv5 packet structure, which can be split into
+three parts:
+
+1. The Handshake data part (up to "Peer IP Address" field)
+2. The HSREQ extension
+3. The KMREQ extension
```

Comment:
I think it would be nice to mark somehow this breakdown on the schema.

Reply:
[ HSv5_PacketStructure_30Apr2019.md.zip](https://github.com/Haivision/srt/files/3132706/HSv5_PacketStructure_30Apr2019.md.zip)